prov/verbs/src/fi_verbs.c

```diff
@@ -454,8 +458,24 @@ int fi_ibv_set_rnr_timer(struct ibv_qp *qp)
 	return 0;
 }
 
+void fi_ibv_cleanup_infra(void)
+{
+	struct fi_ibv_device_info *device_info;
+	struct dlist_entry *tmp_list_entry;
+
+	fi_freeinfo((void *)fi_ibv_verbs_infra.fi_ibv_util_prov.info);
+	fi_ibv_verbs_infra.fi_ibv_util_prov.info = NULL;
+
+	dlist_foreach_container_safe(&fi_ibv_verbs_infra.devices_list,
+				     struct fi_ibv_device_info, device_info,
+				     list_entry, tmp_list_entry) {
+		
+	}
```

Comment:
I'm not following the purpose of this loop.  :)

Reply:
:)
We should free port and device attributes there